shaders/src/common_instancing.glsl

```diff
@@ -0,0 +1,31 @@
+//------------------------------------------------------------------------------
+// Instancing
+// ------------------------------------------------------------------------------------------------
+
+PerRenderableData gObjectUniforms;
+
+void initObjectUniforms(out PerRenderableData p) {
+#if defined(MATERIAL_HAS_INSTANCES)
+    // the material manages instancing, all instances share the same uniform block.
+    p = objectUniforms.data[0];
+#else
+    // automatic instancing was used, each instance has its own uniform block.
```

Comment:
Maybe add a comment here that you're purposefully initializing each field separately.

Reply:
good point. done.